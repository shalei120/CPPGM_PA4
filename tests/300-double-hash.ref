simple ; OP_SEMICOLON
identifier id
simple ; OP_SEMICOLON
literal 1 int 01000000
simple ; OP_SEMICOLON
literal "string" array of 7 char 737472696E6700
simple ; OP_SEMICOLON
literal 'c' char 63
simple ; OP_SEMICOLON
invalid @
simple ; OP_SEMICOLON
simple . OP_DOT
simple ; OP_SEMICOLON
identifier id
simple ; OP_SEMICOLON
literal 1 int 01000000
simple ; OP_SEMICOLON
literal "string" array of 7 char 737472696E6700
simple ; OP_SEMICOLON
literal 'c' char 63
simple ; OP_SEMICOLON
invalid @
simple ; OP_SEMICOLON
simple . OP_DOT
simple ; OP_SEMICOLON
identifier id1id2
simple ; OP_SEMICOLON
simple or_eq OP_BORASS
simple ; OP_SEMICOLON
identifier id11
simple ; OP_SEMICOLON
identifier id11e5
simple ; OP_SEMICOLON
identifier id11_id
simple ; OP_SEMICOLON
literal u8"ab" array of 3 char 616200
simple ; OP_SEMICOLON
literal L"ABC" array of 4 wchar_t 41000000420000004300000000000000
simple ; OP_SEMICOLON
literal R"ab(cd)ab" array of 3 char 636400
simple ; OP_SEMICOLON
literal u'x' char16_t 7800
simple ; OP_SEMICOLON
identifier id1̀
invalid ̀
simple ; OP_SEMICOLON
identifier id1or
simple ; OP_SEMICOLON
user-defined-literal 13_id2 _id2 integer 13
simple ; OP_SEMICOLON
user-defined-literal 1._id2 _id2 floating 1.
simple ; OP_SEMICOLON
user-defined-literal 1.0e+14_id2 _id2 floating 1.0e+14
simple ; OP_SEMICOLON
user-defined-literal 1e-13_id2 _id2 floating 1e-13
simple ; OP_SEMICOLON
user-defined-literal 0.14_id2 _id2 floating 0.14
simple ; OP_SEMICOLON
invalid 1..2_id2
simple ; OP_SEMICOLON
user-defined-literal 0x43_id2 _id2 integer 0x43
simple ; OP_SEMICOLON
literal 0xff int FF000000
simple ; OP_SEMICOLON
literal 0xab13 int 13AB0000
simple ; OP_SEMICOLON
user-defined-literal 0xf_id _id integer 0xf
simple ; OP_SEMICOLON
user-defined-literal "abc"_id2 _id2 string array of 4 char 61626300
simple ; OP_SEMICOLON
user-defined-literal 'c'_id2 _id2 character char 63
simple ; OP_SEMICOLON
identifier and_id2
simple ; OP_SEMICOLON
literal 12 int 0C000000
simple ; OP_SEMICOLON
literal 1.2 double 333333333333F33F
simple ; OP_SEMICOLON
literal 1.2 double 333333333333F33F
simple ; OP_SEMICOLON
literal 1e+15 double 00003426F56B0C43
simple ; OP_SEMICOLON
literal 1e+02 double 0000000000005940
simple ; OP_SEMICOLON
literal 0x034 int 34000000
simple ; OP_SEMICOLON
literal 077 int 3F000000
simple ; OP_SEMICOLON
user-defined-literal 1_̀ _̀ integer 1
simple ; OP_SEMICOLON
literal 1. double 000000000000F03F
simple ; OP_SEMICOLON
invalid 1e+
simple ; OP_SEMICOLON
invalid 3...
simple ; OP_SEMICOLON
literal .14 double EC51B81E85EBC13F
simple ; OP_SEMICOLON
simple |= OP_BORASS
simple ; OP_SEMICOLON
identifier andor
simple ; OP_SEMICOLON
simple >>= OP_RSHIFTASS
simple ; OP_SEMICOLON
eof
