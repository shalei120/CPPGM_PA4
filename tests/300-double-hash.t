#define PLUS(x,y) x ## y
#define ADD(x,y) x ## y
#define SUM(x,y,z) x ## y ## z
PLUS(,);
PLUS(,id);
PLUS(,1);
PLUS(,"string");
PLUS(,'c');
PLUS(,@);
PLUS(,.);
PLUS(id,);
PLUS(1,);
PLUS("string",);
PLUS('c',);
PLUS(@,);
PLUS(.,);

PLUS(id1,id2);
PLUS(or_,eq);
PLUS(id1,1);
PLUS(id1,1e5);
PLUS(id1,1_id);
PLUS(u8,"ab");
PLUS(L,"ABC");
PLUS(R,"ab(cd)ab");
PLUS(u,'x');
PLUS(id1,\u0300\u0300);//id + non-whitespace character that can be part of id
PLUS(id1,or);//id + operator that can be added to id
//id + punctuation can't be valid
PLUS(13,_id2);
PLUS(1.,_id2);
PLUS(1.0e+14,_id2);
PLUS(1e-13,_id2);
PLUS(0.14,_id2);
PLUS(1..2,_id2);//valid ppnumber, invalid number
PLUS(0x43,_id2);
PLUS(0,xff);
PLUS(0x,ab13);
PLUS(0xf,_id);
PLUS("abc",_id2);
PLUS('c',_id2);
//non-whitespace + id can't be valid
PLUS(and,_id2);
//punc + id can't ve valid
PLUS(1,2);
PLUS(1.,2);
PLUS(1,.2);
PLUS(1e+,15);
PLUS(1e+0,2);
PLUS(0x,034);
PLUS(0,77);
//number + string or char do not look valid
PLUS(1_,\u0300);
PLUS(1,.);
//SUM(1e,+,15);if 2nd ## applies first, +15 will be invalid token=>undefined
PLUS(1e,+);//valid pp-number
PLUS(3,...);
PLUS(.,14);
PLUS(|,=);
PLUS(and,or);
PLUS(>,>=);
