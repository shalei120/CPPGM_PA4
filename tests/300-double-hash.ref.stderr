ERROR: invalid floating literal (#1): 1..2
ERROR: invalid floating literal (#2): 1e+
ERROR: invalid floating literal (#1): 3...
